apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerBatchCommand.ExecutableCommand.cs

```diff
@@ -41,6 +41,7 @@ private class ExecutableCommand
             internal SpannerBatchCommandType CommandType { get; }
             internal Priority Priority { get; }
             internal string Tag { get; }
+            internal SpannerConversionOptions ConversionOptions => SpannerConversionOptions.ForConnection(Connection);
```

Comment:
Are we okay to evaluate this one lazily as well? The comment of "The SpannerConnection can be modified, but other objects should be fine" made me somewhat nervous. It's been a while since I looked at SpannerConnection - maybe the bits that can be modified don't include the builder where we get the conversion options? (Or maybe we can just assert that they *shouldn't* change, and behavior isn't guaranteed if they do?)

Reply:
Okay. Sure. I will make it lazy in SpannerCommand.ExecutableCommand as well. 

Just for my understanding - In this class, there is a foreach loop so `SpannerConversionOptions` will be evaluated multiple times. Given that this class has just one internal method and we will always require option in that, is it better to make it lazy or just initialize options in the constructor?

>It's been a while since I looked at SpannerConnection - maybe the bits that can be modified don't include the builder where we get the conversion options? (Or maybe we can just assert that they shouldn't change, and behavior isn't guaranteed if they do?)

I think the builder is immutable. But I didn't understand the action required here. Could you please help.

Reply:
Will look tomorrow - I'm signed out for the day now I'm afraid.

Reply:
I had a look at it again and now I think I understand it a little better.
We have used Lazy initialization now. So, options will be evaluated the first time they are needed. Post that if `ConnectionString` is updated, it will change the builder, but won't be honored as options are already constructed.
How would you like it to be handled?